ui/components/src/model/legend.ts

```diff
@@ -26,7 +26,7 @@ export interface LegendItem {
   label: string;
   isSelected: boolean;
   color: string;
-  onClick: MouseEventHandler<HTMLLIElement>;
+  onClick: MouseEventHandler<HTMLElement>;
```

Comment:
Making this more generic, so it can handle the li and div depending on the virtualization or not.